tensorflow/core/framework/resource_mgr.cc

```diff
@@ -25,6 +25,26 @@ limitations under the License.
 
 namespace tensorflow {
 
+Status setOutputResourceHandle(OpKernelContext* context, int output_index,
+                               const string& container, const string& name,
+                               tensorflow::TypeIndex& type_index) {
+  Tensor* handle;
+  TF_RETURN_IF_ERROR(context->allocate_output(0, TensorShape({}), &handle));
+  ResourceHandle& result = handle->scalar<ResourceHandle>()();
+  result.set_device(context->device()->attributes().name());
```

Comment:
Congrats on your pull request. I think you want to style your code snippets so that anybody that has the same username as one of your local variables doesn't get notifications.